CMakeLists.txt

```diff
@@ -844,39 +844,13 @@ include_directories("include")
 
 ######################################################################################
 
-set(HAVE_DLFCN_H OFF CACHE BOOL "dlopen" FORCE)
-
-if(WIN32)
-  message(STATUS "Using LoadLibrary/FreeLibrary in Windows, libltdl not needed.")
-
-elseif(UNIX)
-
-    find_library(DL_LIB "dl")
-    find_file(DL_H "dlfcn.h")
-
-    if(DL_LIB AND DL_H)
-      message(STATUS "libdl found")
-    else()
-      message(STATUS "libdl not found, assuming dlopen() is in libc")
-      set(DL_LIB "")
-    endif()
-
-    if(DL_H)
-      get_filename_component(DL_H_INCLUDE_DIR "${DL_H}" DIRECTORY)
-      string(FIND "${CMAKE_C_IMPLICIT_INCLUDE_DIRECTORIES}" "${DL_H_INCLUDE_DIR}" LTPOSITION)
-      # include the directory of dlfcn.h, if its not in the default system include dirs
-      # also when cross-compiling this includes <cross-compile-root>/usr/include, which screws things up
```

Comment:
Again worried about cross compiled targets if you remove this code path.

Reply:
when cross-compiling cmake should look in the target prefix and not the native prefix, os I dont understand this logic, and if we want to change that the variable CMAKE_FIND_ROOT_PATH_MODE_INCLUDE can be tweaked to get the desired bahavior so I say this logic is useless

Reply:
OK. Unfortunately xcompile is not regularly tested nor tested automatically, so I trust you on this.